### PR TITLE
refactor build process

### PR DESCRIPTION
**Pull request by McDaddy**
## What type of PR is this?

- [ ] Feature
- [ ] Bugfix
- [ ] Test
- [ ] Documentation
- [x] Refactor
- [ ] Style
- [ ] Chore

## What this PR does / why we need it:
1. remove setup.js in project root, instead this feature will move into erda-ui cli
2. add `erda-ui init` command to install dependency and init .env
3. remove all `.erda/config` dependency
4. replace `ts-loader` with `babel-loader`
5. refactor `build.ts` to enable online build based on new configuration architecture
6. remove copy command
7. optimize some webpack & tsconfig setup
8. add npm script in each package.json, `watch` means dev mode watch, `dev-build` means dev mode build but won't watch
9. TODO launch & i18n script will change later

## Which issue(s) this PR fixes:

Fixes #

## Does this PR introduce a user interface change?

- [ ] Yes(screenshot is required)
- [x] No

### CODE REVIEW

**Comment by McDaddy** [comment]
@Zero-Rock will help to resolve the codecov fail issue

File: cli/lib/copy.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ const { execSync } = child_process;
 export default async (moduleName: string, options: { distPath: string }) => {
```
Comment (daskyrk) [inline cli/lib/copy.ts:23]: This file can be remove?

File: cli/lib/util/env.ts [hunk 2]
```diff
@@ -25,31 +24,37 @@ export const checkIsRoot = () => {
   }
 };
 
-export const getEnvConfig = () => {
-  const { parsed } = dotenv.config({ path: `${process.cwd()}/.env` });
+export const isCwdInRoot = (params?: {currentPath?: string; alert?: boolean}) => {
```
Comment (daskyrk) [inline cli/lib/util/env.ts:27]: missing blank space after and before bracket.

File: cli/lib/util/env.ts [hunk 3]
```diff
@@ -25,31 +24,37 @@ export const checkIsRoot = () => {
...
+  const currentDir = params?.currentPath || process.cwd();
+  let isInRoot = true;
+  if (!fs.existsSync(join(currentDir, 'package.json'))) {
+    isInRoot = false;
+  } else {
+    const pkg = fs.readFileSync(join(currentDir, 'package.json'), 'utf8');
+    const { name } = JSON.parse(pkg);
+    if (name !== 'erda-ui') {
+      isInRoot = false;
```
Comment (daskyrk) [inline cli/lib/util/env.ts:36]: It's better to prompt user the directory name must be `erda-ui`.

File: core/webpack.config.js [hunk 4]
```diff
@@ -196,5 +198,5 @@ module.exports = () => {
     },
   };
 
-  return merge(commonConfig, targetConfig);
+  return smp.wrap(merge(commonConfig, targetConfig));
```
Comment (daskyrk) [inline core/webpack.config.js:201]: This is not used in normal condition?

File: cli/lib/util/env.ts (hunk 3 above)
Comment (McDaddy) [reply cli/lib/util/env.ts:36]: this is not verifying directory name, it just checks whether package.json project name is erda-ui

**Review by daskyrk — APPROVED** [review]

**Pull request merged**